TPP.Persistence.MongoDB/Repos/BadgeMarketRepo.cs

```diff
@@ -0,0 +1,249 @@
+        }
+
+        public async Task<BadgeBuyOffer> CreateBuyOffer(string userId, PkmnSpecies species, string? form, Badge.BadgeSource? source, bool? shiny, int price, int amount, Instant? createdAt = null)
+        {
+            BadgeBuyOffer buyOffer = new BadgeBuyOffer(
+                id: string.Empty,
+                userId: userId,
+                species: species,
+                form: form,
+                source: source,
+                shiny: shiny,
+                price: price,
+                amount: amount,
+                createdAt: createdAt ?? _clock.GetCurrentInstant()
+                );
+
+            await BuyOfferCollection.InsertOneAsync(buyOffer);
+            Debug.Assert(buyOffer.Id.Length > 0, "The MongoDB driver injected a generated ID");
+            return buyOffer;
+        }
+
+        public async Task<Badge> CreateSellOffer(string userId, PkmnSpecies species, string? form, Badge.BadgeSource? source, bool? shiny, int price)
+        {
+            List<Badge> notSellingOwnedByUser = await _badgeRepo.FindAllNotForSaleByCustom(userId, species, form, source, shiny);
+            Badge toSell = SortBySpecialness(notSellingOwnedByUser).First();
+            Badge selling = await _badgeRepo.SetBadgeSellPrice(toSell, price);
+            return selling;
+        }
+
+        public async Task DeleteBuyOffer(string userId, PkmnSpecies species, string? form, Badge.BadgeSource? source, bool? shiny, int amount)
+        {
+            List<BadgeBuyOffer> offers = await FindAllBuyOffers(userId, species, form, source, shiny);
+            if(offers.Count() < amount)
+                throw new ArgumentException(string.Format("Tried to cancel {0} offers but only {1} were found", amount, offers.Count));
+            
+            offers = offers.OrderByDescending(o => o.CreatedAt).ToList();
+            for(int i=0; i < amount; i++)
+            {
+                await BuyOfferCollection.FindOneAndDeleteAsync(o => o.Id == offers[i].Id);
+            }
+        }
+        public async Task DeleteSellOffer(string userId, PkmnSpecies species, string? form, Badge.BadgeSource? source, bool? shiny, int amount)
+        {
+            List<Badge> badgesForSale = await FindAllBadgesForSale(userId, species, form, source, shiny);
+            if (amount > badgesForSale.Count)
+                throw new ArgumentException(string.Format("Tried to cancel {0} offers but only {1} were found", amount, badgesForSale.Count));
+
+            badgesForSale = SortBySpecialness(badgesForSale);
+            badgesForSale.Reverse();
+            for(int i=0; i<amount; i++)
+            {
+                await _badgeRepo.SetBadgeSellPrice(badgesForSale[i], 0);
+            }
+        }
+
+        public async Task<ImmutableList<IBadgeMarketRepo.BadgeSale>> ResolveBuyOffers(PkmnSpecies species, bool? shiny)
+        {
+            List<Badge> badgesForSale = await FindAllBadgesForSale(null, species, null, null, shiny);
```

Comment:
Please add the argument name when it isn't visible, for example `FindAllBadgesForSale(userId: null, species, form: null, source: null, shiny)`